frontend/catalyst/debug/printing.py

```diff
@@ -15,54 +15,90 @@
+
+    One difference is that if the first argument is a string, it may be used similarly to an
+    fstring. Like so:
+
+    .. code-block:: python
+
+        @qjit
+        def cir(a, b, c):
+            debug.print("c={c} b={b} a={a}", a=a, b=b, c=c)
+
+        cir(1, 2, 3)
+    >>> c=3 b=2 a=1
+    """
+
+    if isinstance(fmt, str):
+        debug_callback(partial(_format_print_callback, fmt))(*args, **kwargs)
+        return
+
+    debug_callback(_print_callback)(fmt, *args, **kwargs)
+
+
+def _format_print_callback(fmt: str, *args, **kwargs):
+    """
+    This function has been modified from its original form in the JAX project at
+    https://github.com/google/jax/blob/5eeebf2c1829bf3c66c947b6e12464a779434e29/jax/_src/debugging.py#L269
+    version released under the Apache License, Version 2.0, with the following copyright notice:
+    Copyright 2022 The Jax Authors.
+    """
+    builtins.print(fmt.format(*args, **kwargs))
+
+
+def _print_callback(*args, **kwargs):
+    """Print without formatting"""
+    builtins.print(*args, **kwargs)
+
+
+# pylint: disable=redefined-builtin
+def print_memref(x):
     """A :func:`qjit` compatible print function for printing values at runtime.
 
-    Enables printing of numeric values at runtime. Can also print objects or strings as constants.
+    Enables printing of numeric values at runtime.
 
     Args:
-        x (jax.Array, Any): A single jax array whose numeric values are printed at runtime, or any
-            object whose string representation will be treated as a constant and printed at runtime.
-        memref (Optional[bool]): When set to ``True``, additional information about how the array is
-            stored in memory is printed, via the so-called "memref" descriptor. This includes the
-            base memory address of the data buffer, as well as the rank of the array, the size of
-            each dimension, and the strides between elements.
+        x (jax.Array, Any): A single jax array whose numeric values are printed at runtime.
+            Additional information about how the array is stored in memory is printed.
+            Tensors, in the Catalyst runtime are represented as memref descriptor structs.
+            For more information about memref descirptors visit:
+            https://mlir.llvm.org/docs/Dialects/MemRef/
+            This includes the base memory address of the data buffer, as well as the rank of
+            the array, the size of each dimension, and the strides between elements.
 
     **Example**
 
     .. code-block:: python
 
         @qjit
         def func(x: float):
-            debug.print(x, memref=True)
-            debug.print("exit")
+            debug.print_memref(x)
 
     >>> func(jnp.array(0.43))
     Unranked Memref base@ = 0x5629ff2b6680 rank = 0 offset = 0 sizes = [] strides = [] data =
     [0.43]
-    exit
 
     Outside a :func:`qjit` compiled function the operation falls back to the Python print statement.
```

Comment:
```suggestion
    Outside a :func:`qjit` compiled function the operation falls back to the Python print statement.
    
    .. note::
    
        Python f-strings will not work as expected since they will be treated as Python objects.
        This means that array values embedded in them will have their compile-time representation
        printed, instead of actual data.
        
        For printing format strings at *runtime*, please use :func:`~.debug.print`.
```

Reply:
I feel like this one is no longer relevant since this function is not expected to be compatible with strings nor f-strings?